### ISSUE TITLE
How to read a string written by h5py (python)?

### ISSUE THREAD

**Issue by ggggggggg** [JuliaIO/HDF5.jl#378]
In python I create an HDF5 file with one dataset containing "Hello World".
```python
In [84]: import h5py
In [85]: h5 = h5py.File("test","w")
In [86]: h5["test"]="Hello World"
In [87]: h5.close()
```

Then, in Julia, I try to read that dataset, and get an error.
```julia
julia> using HDF5
julia> h5 = h5open("test","r")
HDF5 data file: test
julia> read(h5["test"])
HDF5-DIAG: Error detected in HDF5 (1.10.0-patch1) thread 0:
  #000: H5Dio.c line 170 in H5Dread(): can't read data
    major: Dataset
    minor: Read failed
  #001: H5Dio.c line 418 in H5D__read(): unable to set up type info
    major: Dataset
    minor: Unable to initialize object
  #002: H5Dio.c line 978 in H5D__typeinfo_init(): unable to convert between src and dest datatype
    major: Dataset
    minor: Feature is unsupported
  #003: H5T.c line 4560 in H5T_path_find(): no appropriate function for conversion path
    major: Datatype
    minor: Unable to initialize object
ERROR: Error reading dataset /test
 in h5d_read(::Int64, ::Int64, ::Int64, ::Int64, ::Int64, ::Array{Ptr{UInt8},1}) at /Users/oneilg/.julia/v0.5/HDF5/src/HDF5.jl:2135
 in read(::HDF5.HDF5Dataset, ::Type{String}) at /Users/user/.julia/v0.5/HDF5/src/HDF5.jl:1288
 in read(::HDF5.HDF5Dataset) at /Users/user/.julia/v0.5/HDF5/src/HDF5.jl:1207
```

Should I be doing something else?

**Comment by timholy** [comment]
Create an H5py.jl repository? HDF5 lets you format "standard types" in many different ways, and HDF5.jl tries not to make any assumptions. It's the foundation for JLD (which supports Julia types) and MAT.jl (which supports Matlab's HDF5 file format); there should be a separate repository to support Python's HDF5 files.

If you browse the issues, the lack of an H5py.jl has oft been pointed out, but no one has ever stepped forward to fill the gap. Since the main people who have developed HDF5/JLD do not seem to also be Python users, it's clearly not going to happen until someone cares enough to do it.

If you decide to tackle this and use MAT as inspiration: it might be good to try for FileIO compatibility. Note that MAT was written before FileIO existed, and has never been updated.

I'm closing this since it's not something for HDF5.jl.

**Comment by andyferris** [comment]
@timholy I thought `h5py` was a python package for writing vanilla HDF5? I ran the python code and get

```
$ h5dump test 
HDF5 "test" {
GROUP "/" {
   DATASET "test" {
      DATATYPE  H5T_STRING {
         STRSIZE H5T_VARIABLE;
         STRPAD H5T_STR_NULLTERM;
         CSET H5T_CSET_ASCII;
         CTYPE H5T_C_S1;
      }
      DATASPACE  SCALAR
      DATA {
      (0): "Hello World"
      }
   }
}
}
```

There was a known issue (#343) with string encodings where `h5py` (or anything else) writes ascii encoded strings and `HDF5.jl` can only read utf-8 encoded strings (it used to work in Julia 0.4, but not 0.5)... 

I think supporting the standard HDF5 ASCII encoded string datatype should be something for HDF5.jl.

**Comment by andyferris** [comment]
PS - I did submit a PR to make this work at #340 (but unfortunately, it didn't receive any attention from the maintainers).

**Comment by andyferris** [comment]
@ggggggggg despite what I said, this issue should probably remain closed since it is a duplicate of #332.

**Comment by andyferris** [comment]
@ggggggggg I just rebased #340 so you should be able to use that branch, if it's important.
